Bug 2293881: osd: revert limit storageClassDeviceSets to 63 chars

subhamkrai opened the pull request:
This commits reverts changes done to limit the storageClassDeviceSets Name for 4.16 due to upgrade issues.

There is another commit added in this pr `build: remove iproute build dependency on centos repo` so that most of the build and e2e CI is green

<!-- Thank you for contributing to Rook! -->

<!-- STEPS TO FOLLOW:
  1. Add a description of the changes (frequently the same as the commit description)
  2. Enter the issue number next to "Resolves #" below (if there is no tracking issue resolved, **remove that section**)
  3. Review our Contributing documentation at https://rook.io/docs/rook/latest/Contributing/development-flow/
  4. Follow the steps in the checklist below, starting with the **Commit Message Formatting**.
-->

<!-- Uncomment this section with the issue number if an issue is being resolved
**Issue resolved by this Pull Request:**
Resolves #
--->

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

subhamkrai commented:
/bugzilla refresh

agarwal-mudit reviewed (approved):
/lgtm

subhamkrai commented:
> /lgtm

added one more comment to make most CI happy  @agarwal-mudit 

cc @sp98

agarwal-mudit commented:
/lgtm

SanjalKatiyar commented:
@subhamkrai general question: I can see we added validation in CephCluster CR, but downstream this is controlled by StorageCluster CR. Shouldn't we do the same for StorageCluster CRD as well ??

subhamkrai commented:
> @subhamkrai general question: I can see we added validation in CephCluster CR, but downstream this is controlled by StorageCluster CR. Shouldn't we do the same for StorageCluster CRD as well ??

stroageCluster Name will come to rook only and in rook it was throwing error, so we have added here

subhamkrai commented:
@agarwal-mudit @sp98  ^^

sp98 commented:
/lgtm
